### PR TITLE
Enforce EQ pressure minimal reduction at service rate

### PR DESCRIPTION
**Pull request by cesarBLG**
Currently the "minimal reduction" pressure drop is performed instantly. With this changes the pressure drops at MaxApplicationRate as expected, and ensures that the valve is not lapped before completing the minimal reduction.

See http://www.elvastower.com/forums/index.php?/topic/34853-equalizing-reservoir-unexpected-behavior/

Bug report: https://bugs.launchpad.net/or/+bug/1915432

### CODE REVIEW

**Comment by cesarBLG** [comment]
I've just noticed a potential issue with vacuum brakes, so I have to address it before adding it to unstable

File: Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Controllers/MSTSBrakeController.cs [hunk 1]
```diff
@@ -104,6 +106,17 @@ public override void UpdatePressure(ref float pressureBar, float elapsedClockSec
...
+
+                    if (type == ControllerState.Hold || type == ControllerState.Lap || type == ControllerState.MinimalReduction)
+                    {
+                        if (EnforceMinimalReduction)
+                            DecreasePressure(ref pressureBar, MaxPressureBar() - MinReductionBar(), ApplyRateBarpS(), elapsedClockSeconds);
+                    }
+                    else
+                    {
+                        EnforceMinimalReduction = false;
+                    }
```
Comment (YoRyan) [inline Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Controllers/MSTSBrakeController.cs:118]:
```suggestion
                    switch (type)
                    {
                        case ControllerState.Hold:
                        case ControllerState.Lap:
                        case ControllerState.MinimalReduction:
                            if (EnforceMinimalReduction)
                                DecreasePressure(ref pressureBar, MaxPressureBar() - MinReductionBar(), ApplyRateBarpS(), elapsedClockSeconds);
                            break;
                        default:
                            EnforceMinimalReduction = false;
                            break;
                    }
```
With three or more cases for the same variable, I suggest using a switch-case.

**Review by YoRyan — APPROVED** [review]
Thanks, this looks good to me. I'll approve, but let's wait a week until merging so that we know there haven't been any unintended changes.

**Comment by cesarBLG** [comment]
Thanks, Ryan

**Pull request merged**